rust_src/remacs-sys/lib.rs

```diff
@@ -1247,6 +1247,7 @@ extern "C" {
     pub fn window_box_left_offset(w: *const Lisp_Window, area: glyph_row_area) -> c_int;
     pub fn window_menu_bar_p(w: *const Lisp_Window) -> bool;
     pub fn window_tool_bar_p(w: *const Lisp_Window) -> bool;
+    pub fn recenter_overlay_lists(buffer: *mut Lisp_Buffer, pos: ptrdiff_t);
```

Comment:
Did you want to give porting this function a shot?

Reply:
Good idea. If I manage to rewrite the original in Rust, I'll try to benchmark both versions, too.

Reply:
:+1:

Reply:
`LispOverlayRef` is in rust_src/src/buffer.rs. That is a good place to start. The start and end methods replace the `OVERLAY_START` and `OVERLAY_END` calls in the C code.

Reply:
While reading the source for this code, I found [this](https://github.com/Wilfred/remacs/blob/master/etc/TODO#L263), which suggests that `recenter_overlay_lists` will be reimplemented in another way. Do you think it still makes sense to rewrite it in Rust, given that?

Reply:
That is a emacs TODO, not ours. Just FYI.

If you look at the upstream tree there is a branch for hacking on this. https://github.com/emacs-mirror/emacs/commits/feature/noverlay/src

Note that once we have finished porting Emacs to Rust we will be free to reimplement these internals.